avm/res/compute/virtual-machine/main.bicep

```diff
@@ -120,37 +120,8 @@ param backupVaultResourceGroup string = resourceGroup().name
 @description('Optional. Backup policy the VMs should be using for backup. If not provided, it will use the DefaultPolicy from the backup recovery service vault.')
 param backupPolicyName string = 'DefaultPolicy'
 
-@description('Optional. The status of the schedule (i.e. Enabled, Disabled).')
-@allowed([
-  'Enabled'
-  'Disabled'
-])
-param autoShutdownStatus string = 'Disabled'
-
-@description('Optional. The time of day the schedule will occur.')
-param autoShutdownTime string = '19:00'
-
-@description('Optional. The time zone id.')
-param autoShutdownTimeZone string = 'UTC'
-
-@description('Optional. If notifications are enabled for this schedule (i.e. Enabled, Disabled).')
-@allowed([
-  'Enabled'
-  'Disabled'
-])
-param autoShutdownNotificationStatus string = 'Disabled'
-
-@description('Optional. The locale to use when sending a notification (fallback for unsupported languages is EN).')
-param autoShutdownNotificationLocale string = 'en'
-
-@description('Optional. The email recipient to send notifications to (can be a list of semi-colon separated email addresses).')
-param autoShutdownNotificationEmail string = ''
-
-@description('Optional. The webhook URL to which the notification will be sent.')
-param autoShutdownWebhookUrl string = ''
-
-@description('Optional. Time in minutes before event at which notification will be sent.')
-param autoShutdownNotificationTimeInMinutes int = 30
+@description('Optional. The configuration for auto-shutdown.')
+param autoShutdownConfig object = {}
```

Comment:
It would be ideal to define complex objects like this with user defined types.
I'll not block the PR on that, but just for awareness. @rahalan you can resolve once read.

Reply:
@eriqua I know, will do in a later iteration